src/components/input/input.js

```diff
@@ -480,6 +530,7 @@ function placeholderDirective($log) {
   }
 }
 
+var visibilityDirectives = ['ngIf', 'ngShow', 'ngHide', 'ngSwitchWhen', 'ngSwitchDefault'];
```

Comment:
It might be worth mentioning in the docs that this understands this set of directives and that any custom hide/show directives won't behave the same.

Do we need to add things like `hide-gt-sm` ?